mpld3/js/mpld3.v0.3git.js

```diff
@@ -580,9 +581,24 @@
     delete pathProps.color;
     pathProps.edgewidth = pathProps.linewidth;
     delete pathProps.linewidth;
+    drawstyle = pathProps.drawstyle;
+    delete pathProps.drawstyle;
     this.defaultProps = mpld3_Path.prototype.defaultProps;
     mpld3_Path.call(this, ax, pathProps);
-    this.datafunc = d3.svg.line().interpolate("linear");
+    alert(drawstyle);
+    switch (drawstyle) {
+     case "steps":
+     case "steps-pre":
+      this.datafunc = d3.svg.line().interpolate("step-before");
+      break;
+
+     case "steps-post":
+      this.datafunc = d3.svg.line().interpolate("step-after");
+      break;
+
```

Comment:
My d3 patch attempt didn’t go well, but it was late and Javascript is not my preferred language.
I will have another go and as you say its a clear two lines here if I put the support in d3.

It could be hacked in here in full but that would be less elegant and useful.
Good to know there is demand for it.

> On 23 Mar 2016, at 12:37 AM, Abraham Flaxman <notifications@github.com> wrote:
> 
> In mpld3/js/mpld3.v0.3git.js <https://github.com/jakevdp/mpld3/pull/350#discussion_r56986081>:
> 
> >      this.defaultProps = mpld3_Path.prototype.defaultProps;
> >      mpld3_Path.call(this, ax, pathProps);
> > -    this.datafunc = d3.svg.line().interpolate("linear");
> > +    alert(drawstyle);
> > +    switch (drawstyle) {
> > +     case "steps":
> > +     case "steps-pre":
> > +      this.datafunc = d3.svg.line().interpolate("step-before");
> > +      break;
> > +
> > +     case "steps-post":
> > +      this.datafunc = d3.svg.line().interpolate("step-after");
> > +      break;
> > +
> There have been some requests for steps-mid support, and I bet this is just where it would be added. I know it's harder than steps-pre and steps-post, so no need to add it now... I'm just trying to plan ahead, while this is fresh in your mind.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/jakevdp/mpld3/pull/350/files/97d8478e21d60ba02d9b669b15650fdee984876b#r56986081>


______________________________________________________________________

The information in this email is confidential and intended solely for the addressee.
You must not disclose, forward, print or use it without the permission of the sender.
______________________________________________________________________